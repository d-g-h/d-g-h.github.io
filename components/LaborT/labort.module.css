.labor {
  display: flex;
  flex-direction: column;
  margin-bottom: 0.75rem;
}

.header {
  display: flex;
  flex-direction: column;
  align-items: baseline;
  justify-content: space-between;

  @media (min-width: 640px) {
    flex-direction: row;
  }
}

.group {
  display: flex;
  align-items: baseline;
}

.title {
  font-size: 1.25rem;
  font-weight: bold;
}
.highlight {
  color: var(--lime-color);
  font-size: 0.75rem;
  margin-left: 0.25rem;
}
.time {
  font-size: 0.75rem;
  margin-bottom: 0.25rem;
  text-align: right;
}
.location {
  @media (min-width: 640px) {
    text-align: right;
  }
  font-size: 0.75rem;
}

@media print {
  .highlight {
    color: var(--dark-gray-color);
  }
}
