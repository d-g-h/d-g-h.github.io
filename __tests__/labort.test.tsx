/**
 * @jest-environment jsdom
 */
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import LaborT from "@/components/labort";

const KEY = "0";
const COMPANY = "Company";
const TITLE = "Job Title";
const LOCATION = "Location";
const START = "2020-02";
const END = "2024-02";

describe("Labor", () => {
  it("renders Labor", () => {
    render(
      <LaborT
        key={KEY}
        title={TITLE}
        location={LOCATION}
        company={COMPANY}
        start={START}
        end={END}
      >
        <></>
      </LaborT>
    );

    const jobTitle = screen.getByText("Job Title");
    const location = screen.getByText("Location");

    expect(jobTitle).toBeInTheDocument();
    expect(location).toBeInTheDocument();
  });
  it("renders Labor without Company", () => {
    render(
      <LaborT
        key={KEY}
        title={TITLE}
        location={LOCATION}
        start={START}
        end={END}
      >
        <></>
      </LaborT>
    );

    const jobTitle = screen.getByText("Job Title");
    const company = screen.queryByText("Company");

    expect(jobTitle).toBeInTheDocument();
    expect(company).toBeNull();
  });
  it("renders html correctly", () => {
    const { container } = render(
      <LaborT
        key={KEY}
        title={TITLE}
        location={LOCATION}
        company={COMPANY}
        start={START}
        end={END}
      >
        <></>
      </LaborT>
    );
    expect(container).toMatchSnapshot();
  });
});
