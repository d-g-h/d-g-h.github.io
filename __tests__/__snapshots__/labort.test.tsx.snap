// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Labor renders html correctly 1`] = `
<div>
  <li
    class="labor"
  >
    <div
      class="header"
    >
      <div
        class="group"
      >
        <div
          class="title"
        >
          Job Title
        </div>
        <span>
          ,
        </span>
        <div
          class="highlight"
        >
          Company
        </div>
      </div>
      <div>
        <div
          class="location"
        >
          Location
        </div>
        <div
          class="time"
        >
          <time
            datetime="2020-02"
          >
            February 2020
          </time>
          <span>
             - 
          </span>
          <time
            datetime="2024-02"
          >
            February 2024
          </time>
        </div>
      </div>
    </div>
  </li>
</div>
`;
